src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int FIGURES_TO_GENERATE = 10;
+
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < FIGURES_TO_GENERATE; i++) {
+            Figure currentFigure;
+            if (i < 5) {

```

Comment:
if we change value of `FIGURES_TO_GENERATE`, your app would work incorrect. please calculate that value dynamically, depending on `FIGURES_TO_GENERATE` value